rust/tools/authorization-logic/raksha_examples/policy_patterns/NonUnanimousDecisions

```diff
@@ -0,0 +1,128 @@
+    // though the decision might not be DataOwner1's preference because
+    // opting in might be the only way to use this service.
+    "RoleBasedDecisionPolicy" canSay 
+        "ExampleParticle" may("send_remotely", "datum1").
+                // (this names a specific particle, but for real policies,
+                // this can be made generic).
+
+    // DataOwner1 prefers that data depending on datum1 is not sent remotely
+    // (This syntax is an attribute as described 
+    // [here](https://github.com/aferr/raksha/tree/main/rust/tools/authorization-logic#attributes-and-canactas)).
+    "DataOwner1" hasPreference("do_not_send_remotely").
+}
+
+"DataOwner2" says {
+    owns("datum2").
+    
+    "RoleBasedDecisionPolicy" canSay 
+        "ExampleParticle" may("send_remotely", "datum2").
+   
+    // Data Owner1 prefers that data depending on datum1 is not sent remotely
+    "DataOwner2" hasPreference("do_send_remotely").
+}
+
+// This is a principal that is trusted by "RoleBasedDecisonPolicy" to
+// determine if another principal can occupy a particular role. As in
+// role-based access control, whether or not a principal occupies a role
+// can be used to make access decisions. In this example the role is
+// "DeviceOwner". Other roles are also possible, and it is also possible
+// to make membership of a role conditional on other facts.
+"RoleAuthority" says {
+    "DataOwner2" canActAs "DeviceOwner".
+}
+
+
+// This is a policy that collects the preferences of both data owners
+// on whether or not they are OK with data being sent. However, the 
+// actual decision depends only on the preference of the principal
+// in a specifc role (DeviceOwner).
+
+// This role-based decision policy is one way of resolving the conflict,
+// but the pattern of delegating 
+"RoleBasedDecisionPolicy" says {
+    // The "RoleAuthority" is trusted by the policy to decide who occupies
+    // the device owner role.
+    "RoleAuthority" canSay PrincipalX canActAs "DeviceOwner" :-
+        isPrincipal(PrincipalX).
+
+    // If there is a handle that depends on data from BOTH "datum1" and
+    // "datum2", in order to permit the particle to "send_remotely"
+    // (i.e. pass the check generated by "will("send_remotely", handleX),
+    // we need to prove BOTH `"DO1" says may("send_remotely", "datum1")`
+    // AND `"DO2" says may("send_remotely", "datum2"). Because both owners
+    // have delegated the right to make this decision to this policy, both
+    // owners will respect the outcome from the policy.
+  
+    // In this case, the preference of the device owner is the only one
+    // that matters for both pieces of data.
+    "ExampleParticle" may("send_remotely", "datum1") :-
+        PrincipalX canActAs "DeviceOwner",
+        PrincipalX hasPreference("do_send_remotely").
+    "ExampleParticle" may("send_remotely", "datum2") :-
+        PrincipalX canActAs "DeviceOwner",
+        PrincipalX hasPreference("do_send_remotely").
```

Comment:
I'm reading these examples as

"""
ExampleParticle may send `datum(1|2)` remotely if there exists an arbitrary principal, `PrincipalX`, that can act as `DeviceOwner` and that arbitrary principal has the `do_send_remotely` preference.
"""

Is that right? It seems like there's a disconnect in the `Principal` that the head and body of the rule are discussing. Should `ExampleParticle` be replaced with `PrincipalX` or `PrincipalX` be replaced with `ExampleParticle`?

Reply:
I think your policy is definitely more robust for the case where there are multiple device owners (or in general multiple parties that can occupy the relevant role). 

Since this is a piece of code meant to demo a pattern, I prefer the simpler version, since it might be easier to understand. I added a comment that clarifies that here it's assumed there's just one device owner.

Reply:
Sounds good, simpler is probably better.